### DISCUSSION TITLE
Split array on N arrays

### DISCUSSION THREAD

**Discussion by adinvadim** [gcanti/fp-ts#1653] [Q&A]
May be is there in fp-ts/array util, that split array on N arrays.
I found chunksOf, but it is split on arrays with length N

**Comment by ammut** [comment]
There probably isn't, but its rather easy to do:

```
export const divide: (n: number) => <A>(as: Array<A>) => Array<NonEmptyArray<A>> = (n) => (as) => {
  const length = Math.ceil(as.length / n);
  return A.chunksOf(length);
};
```

However, I strongly suggest you write up a few unit tests for this function so you are aware of what happens if `n` doesn't divide `as.length` evenly. For example, what do think the result of `divide(2)([1, 2, 3])` should be? Or what about `divide(2)([1])`? Or `divide(2)([])`?

<details>
  <summary>(answer)</summary>

  The results are `[[1, 2], [3]]`, `[[1]]`, and `[]`
</details>
